src/System.Net.NetworkInformation/src/System/Net/NetworkInformation/LinuxIPGlobalProperties.cs

```diff
@@ -0,0 +1,362 @@
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System.Globalization;
+using System.IO;
+
+namespace System.Net.NetworkInformation
+{
+    internal class LinuxIPGlobalProperties : IPGlobalProperties
+    {
+        private static readonly string[] newLineSeparator = new string[] { Environment.NewLine }; // Used for string splitting
+
+        public override string DhcpScopeName
+        {
+            get
+            {
+                throw new NotImplementedException();
+            }
+        }
+
+        public override string DomainName
+        {
+            get
+            {
+                return HostInformation.DomainName;
+            }
+        }
+
+        public override string HostName
+        {
+            get
+            {
+                return HostInformation.HostName;
+            }
+        }
+
+        public override bool IsWinsProxy
+        {
+            get
+            {
+                return false;
+            }
+        }
+
+        public override NetBiosNodeType NodeType
+        {
+            get
+            {
+                // Could return NetBiosNodeType.Unknown (?)
+                throw new NotImplementedException();
+            }
+        }
+
+        public override TcpConnectionInformation[] GetActiveTcpConnections()
+        {
+            string tcp4FileContents = File.ReadAllText(LinuxNetworkFiles.Tcp4ConnectionsFile);
+            string[] v4connections = tcp4FileContents.Split(newLineSeparator, StringSplitOptions.RemoveEmptyEntries);
+
+            string tcp6FileContents = File.ReadAllText(LinuxNetworkFiles.Tcp6ConnectionsFile);
+            string[] v6connections = tcp6FileContents.Split(newLineSeparator, StringSplitOptions.RemoveEmptyEntries);
+
+            TcpConnectionInformation[] connections = new TcpConnectionInformation[v4connections.Length + v6connections.Length - 2]; // First line is header in each file
+            int index = 0;
+
+            // UDP Connections
+            for (int i = 1; i < v4connections.Length; i++) // Skip first line header
+            {
+                string line = v4connections[i];
+                connections[index++] = ParseTcpConnectionInformationFromLine(line);
+            }
+
+            // UDP6 Connections
+            for (int i = 1; i < v6connections.Length; i++) // Skip first line header
+            {
+                string line = v6connections[i];
+                IPAddress remoteIPAddress;
+                int remotePort;
+                ParseRemoteConnectionInformation(line, out remoteIPAddress, out remotePort);
+
+                connections[index++] = ParseTcpConnectionInformationFromLine(line);
+            }
+
+            return connections;
+        }
+
+        public override IPEndPoint[] GetActiveTcpListeners()
+        {
+            string tcp4FileContents = File.ReadAllText(LinuxNetworkFiles.Tcp4ConnectionsFile);
```

Comment:
Optimization note: the data can be cached to avoid multiple reads of the same file in a tight loop. Most likely the kernel won't update the file faster than a few hundred milliseconds anyway.